_aggregations/bucket/composite.md

```diff
@@ -0,0 +1,826 @@
+          },
+          "doc_count": 7
+        }
+      ]
+    }
+  }
+}
+```
+
+## Requesting sub-aggregations
+
+Composite aggregations are most useful when combined with sub-aggregations that reveal information about the documents in the `composite` buckets.
+
+### Example: Sub-aggregations
+
+The following example request compares average spending based on gender for each day of the week in the OpenSearch Dashboards sample e-commerce data:
+
+```json
+GET opensearch_dashboards_sample_data_ecommerce/_search
+{
+  "size": 0,
+  "aggs": {
+    "composite_buckets": {
+      "composite": {
+        "sources": [
+          { "weekday": { "terms": { "field": "day_of_week" }}},
+          { "gender": { "terms": { "field": "customer_gender" }}}          
+        ],
+        "size": 6
+      },
+      "aggs": {
+        "avg_spend": {
+          "avg": { "field": "taxful_total_price" }
+        }
+      }
+    }
+  }
+}
+```
+
+### Example results: Sub-aggregations
+
+As shown in the following example response, the aggregation returns the average `taxful_total_price` of the first `6` buckets:
+
+```json
+{
+  "took": 30,
+  "timed_out": false,
+  "_shards": {
+    "total": 1,
+    "successful": 1,
+    "skipped": 0,
+    "failed": 0
+  },
+  "hits": {
+    "total": {
+      "value": 4675,
+      "relation": "eq"
+    },
+    "max_score": null,
+    "hits": []
+  },
+  "aggregations": {
+    "composite_buckets": {
+      "after_key": {
+        "weekday": "Saturday",
+        "gender": "MALE"
+      },
+      "buckets": [
+        {
+          "key": {
+            "weekday": "Friday",
+            "gender": "FEMALE"
+          },
+          "doc_count": 399,
+          "avg_spend": {
+            "value": 71.7733395989975
+          }
+        },
+        {
+          "key": {
+            "weekday": "Friday",
+            "gender": "MALE"
+          },
+          "doc_count": 371,
+          "avg_spend": {
+            "value": 79.72514108827494
+          }
+        },
+        {
+          "key": {
+            "weekday": "Monday",
+            "gender": "FEMALE"
+          },
+          "doc_count": 320,
+          "avg_spend": {
+            "value": 72.1588623046875
+          }
+        },
+        {
+          "key": {
+            "weekday": "Monday",
+            "gender": "MALE"
+          },
+          "doc_count": 259,
+          "avg_spend": {
+            "value": 86.1754946911197
+          }
+        },
+        {
+          "key": {
+            "weekday": "Saturday",
+            "gender": "FEMALE"
+          },
+          "doc_count": 365,
+          "avg_spend": {
+            "value": 73.53236301369863
+          }
+        },
+        {
+          "key": {
+            "weekday": "Saturday",
+            "gender": "MALE"
+          },
+          "doc_count": 371,
+          "avg_spend": {
+            "value": 72.78092360175202
+          }
+        }
+      ]
+    }
+  }
+}
+```
+
+## Paginating composite results
```

Comment:
How long are results cached for display pagination? 

Are there any shortcuts to displaying result pages, or do you have to give it the entire request plus `after` parameter each time?